File: swiss-service/src/main/java/br/com/vsconsulting/swiss/service/ProjectService.java
package br.com.vsconsulting.swiss.service;

import java.util.Date;

import br.com.vsconsulting.swiss.respository.ProjectRepository;
import br.com.vsconsulting.swiss.to.ProjectSearchTo;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Sort;
import org.springframework.data.domain.Sort.Direction;
import org.springframework.data.domain.Sort.Order;
import org.springframework.data.mongodb.core.MongoTemplate;
import org.springframework.stereotype.Service;

import br.com.vsconsulting.swiss.model.entities.Project;

@Service
public class ProjectService {

	@Autowired
	private ProjectRepository repository;

	public Project load(String id) {
		return this.repository.findOne(id);
	}
	
	public Project save(Project project) {
		return this.repository.save(project);
	}
	
	public Page<Project> search(ProjectSearchTo searchTo) {
		if(searchTo == null) {
			return null;
		}		
		if(searchTo.getSearchTerm() == null) {
			searchTo.setSearchTerm("");
		}
		if(searchTo.getSort() == null) {
			searchTo.setSort(new Sort(new Order(Direction.ASC, "title")));
		}
		return this.repository.search(searchTo.getSearchTerm(), searchTo.isOnlyActives(), searchTo);
	}

	public Iterable<Project> listAll() {
		return this.repository.findAll();
	}

	public Project archive(Project project) {
		if(project == null || project.getArchivationDate() != null) {
			return project;
		}
		project.setArchivationDate(new Date());
		return this.save(project);
	}

	public Project activate(Project project) {
		if(project == null || project.getArchivationDate() == null) {
			return project;
		}
		project.setArchivationDate(null);
		return this.save(project);
	}
	
}

File: swiss-view/src/app/swiss/projects/project.service.js
(function() {
    'use strict';

    angular
        .module('swiss-module')
        .factory('projectService', ProjectService);


    /* @ngInject */
    function ProjectService($http) {
    	var urlBase = '/swiss/rest/project' ;
        var service = {
            activate: activate,
        	archive: archive,
            save: save,
            load: load,
            remove: remove,
            find: find,
            list: list
        };
        return service;

        function save(project) {
        	return $http.post(urlBase+'/save', project);
        };

        function archive(project) {
        	return $http.post(urlBase+'/archive', project);
        };

        function activate(project) {
        	return $http.post(urlBase+'/activate', project);
        };

        function load (id) {
        	return $http.put(urlBase+'/load/' + id);
        };

        function remove (id) {
            return $http.delete(urlBase + '/delete/' + id);
        };

        function list () {
            return $http.get(urlBase + '/list');
        };

        function find (searchObject) {
            return $http.post(urlBase + '/find', searchObject);
        };
    }
})();
